sql/catalyst/src/test/scala/org/apache/spark/sql/catalyst/expressions/aggregate/DeclarativeAggregateEvaluator.scala

```diff
@@ -17,24 +17,24 @@
 package org.apache.spark.sql.catalyst.expressions.aggregate
 
 import org.apache.spark.sql.catalyst.InternalRow
-import org.apache.spark.sql.catalyst.expressions.{Attribute, JoinedRow, SafeProjection}
```

Comment:
`SafeProjection` works fine until the aggregation buffer contains more than one field. In that case, if the second expression in the function's `updateExpressions` looks back at the buffer's first field, that first field may have already be changed by the first expression.

`MutableExpression`, on the other hand, keeps an intermediate results row, so the current buffer is not affected until all expressions in `updateExpressions` have been evaluated.